test/functional/invalidtxrequest.py

```diff
@@ -8,55 +8,46 @@
 import time
 
 from test_framework.blocktools import create_block, create_coinbase, create_transaction
-from test_framework.comptool import RejectResult, TestInstance, TestManager
 from test_framework.messages import COIN
-from test_framework.mininode import network_thread_start
-from test_framework.test_framework import ComparisonTestFramework
+from test_framework.mininode import network_thread_start, P2PStub
+from test_framework.test_framework import BitcoinTestFramework
 
-class InvalidTxRequestTest(ComparisonTestFramework):
+class InvalidTxRequestTest(BitcoinTestFramework):
 
     def set_test_params(self):
         self.num_nodes = 1
         self.setup_clean_chain = True
+        self.extra_args = [["-whitelist=127.0.0.1"]]
 
     def run_test(self):
-        test = TestManager(self, self.options.tmpdir)
-        test.add_all_connections(self.nodes)
-        self.tip = None
-        self.block_time = None
+        # Add p2p connection to node0
+        node = self.nodes[0]  # convenience reference to the node
+        node.add_p2p_connection(P2PStub())
+
         network_thread_start()
-        test.run()
+        node.p2p.wait_for_verack()
 
-    def get_tests(self):
-        self.tip = int("0x" + self.nodes[0].getbestblockhash(), 0)
-        self.block_time = int(time.time()) + 1
+        tip = int("0x" + self.nodes[0].getbestblockhash(), 0)
+        block_time = int(time.time()) + 1
 
         self.log.info("Create a new block with an anyone-can-spend coinbase.")
         height = 1
-        block = create_block(self.tip, create_coinbase(height), self.block_time)
-        self.block_time += 1
+        block = create_block(tip, create_coinbase(height), block_time)
+        block_time += 1
         block.solve()
         # Save the coinbase for later
-        self.block1 = block
-        self.tip = block.sha256
+        block1 = block
+        tip = block.sha256
         height += 1
-        yield TestInstance([[block, True]])
+        node.p2p.send_blocks_and_test([block], node, True)
 
         self.log.info("Mature the block.")
-        test = TestInstance(sync_every_block=False)
-        for i in range(100):
-            block = create_block(self.tip, create_coinbase(height), self.block_time)
-            block.solve()
-            self.tip = block.sha256
-            self.block_time += 1
-            test.blocks_and_transactions.append([block, True])
-            height += 1
-        yield test
+        self.nodes[0].generate(100)
 
         # b'\x64' is OP_NOTIF
         # Transaction will be rejected with code 16 (REJECT_INVALID)
-        tx1 = create_transaction(self.block1.vtx[0], 0, b'\x64', 50 * COIN - 12000)
-        yield TestInstance([[tx1, RejectResult(16, b'mandatory-script-verify-flag-failed')]])
+        tx1 = create_transaction(block1.vtx[0], 0, b'\x64', 50 * COIN - 12000)
+        node.p2p.send_txs_and_test([tx1], node, False, 16, b'mandatory-script-verify-flag-failed (Invalid OP_IF construction)')
```

Comment:
In commit "Change invalidtxrequest to use BitcoinTestFramework"

Maybe add `success=`, `reject_code=`, keywords for future proofing and clarity.

Reply:
yes